src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -1059,12 +1059,20 @@ private static IEnumerable<SelectListItem> GetSelectListItems([NotNull] ViewCont
             values = values.Concat(enumValues);
 
             selectedValues = new HashSet<string>(values, StringComparer.OrdinalIgnoreCase);
+
+            // Perform deep copy of selectList to avoid changing user's Selected property values.
             var newSelectList = new List<SelectListItem>();
             foreach (SelectListItem item in selectList)
             {
-                item.Selected =
-                    (item.Value != null) ? selectedValues.Contains(item.Value) : selectedValues.Contains(item.Text);
-                newSelectList.Add(item);
+                var newItem = new SelectListItem
```

Comment:
if we need to dupe this code anywhere, sure. right now this is the only instance and `SelectListItem` contains only auto-properties, no explicit constructor.